params/astria_config.go

```diff
@@ -0,0 +1,444 @@
+				log.Warn("fee asset collectors not set, assets will be burned", "fork", fork.Name)
+			}
+
+			for _, pType := range fork.Precompiles {
+				if err := pType.Validate(); err != nil {
+					return fmt.Errorf("fork %s: invalid precompile %s", fork.Name, pType)
+				}
+			}
+		} else {
+			log.Warn("fork will halt", "fork", fork.Name, "height", fork.Height)
+		}
+	}
+
+	return nil
+}
+
+func GetDefaultAstriaForkData() AstriaForkData {
+	return AstriaForkData{
+		Name:         "default",
+		Height:       1,
+		FeeCollector: common.Address{},
+		EIP1559Params: AstriaEIP1559Params{
+			MinBaseFee:               0,
+			ElasticityMultiplier:     DefaultElasticityMultiplier,
+			BaseFeeChangeDenominator: DefaultBaseFeeChangeDenominator,
+		},
+		BridgeAddresses:     make(map[string]*AstriaBridgeAddressConfig),
+		BridgeAllowedAssets: make(map[string]struct{}),
+		Precompiles:         make(map[common.Address]PrecompileType),
+	}
+}
+
+func (c *AstriaForks) GetForkAtHeight(height uint64) AstriaForkData {
+	idx := sort.Search(len(c.orderedForks), func(i int) bool {
+		return c.orderedForks[i].Height <= height
+	})
+	// no named fork at this height
+	if idx == len(c.orderedForks) {
+		return GetDefaultAstriaForkData()
+	}
+	return c.orderedForks[idx]
+}
+
+func (c *AstriaForks) GetNextForkAtHeight(height uint64) *AstriaForkData {
+	// orderedForks are sorted in descending order; iterate from the end
+	for i := len(c.orderedForks) - 1; i >= 0; i-- {
+		if c.orderedForks[i].Height > height {
+			return &c.orderedForks[i]
+		}
+	}
+	return nil
+}
+
+func (c *AstriaForks) MinBaseFeeAt(height uint64) *big.Int {
+	return big.NewInt(0).SetUint64(c.GetForkAtHeight(height).EIP1559Params.MinBaseFee)
+}
+
+// BaseFeeChangeDenominator bounds the amount the base fee can change between blocks.
+func (c *AstriaForks) BaseFeeChangeDenominatorAt(height uint64) uint64 {
+	return c.GetForkAtHeight(height).EIP1559Params.BaseFeeChangeDenominator
+}
+
+// ElasticityMultiplier bounds the maximum gas limit an EIP-1559 block may have.
+func (c *AstriaForks) ElasticityMultiplierAt(height uint64) uint64 {
+	return c.GetForkAtHeight(height).EIP1559Params.ElasticityMultiplier
+}
+
+func (c *AstriaForks) MarshalJSON() ([]byte, error) {
+	return json.Marshal(c.forkMap)
+}
+
+func (c *AstriaForks) UnmarshalJSON(data []byte) error {
+	var forkMap map[string]AstriaForkConfig
+	if err := json.Unmarshal(data, &forkMap); err != nil {
+		return err
+	}
+
+	newForks, err := NewAstriaForks(forkMap)
+	if err != nil {
+		return err
+	}
+
+	*c = *newForks
+	return nil
+}
+
+func (c *ChainConfig) GetAstriaForks() *AstriaForks {
+	if c.AstriaForks == nil {
+		forks, _ := NewAstriaForks(nil)
+		return forks
+	}
+	return c.AstriaForks
+}
+
+type AstriaBridgeAddressConfig struct {
+	BridgeAddress  string                  `json:"bridgeAddress"`
+	SenderAddress  common.Address          `json:"senderAddress,omitempty"`
+	StartHeight    uint32                  `json:"startHeight"`
```

Comment:
do we need a `startHeight` here? Since we already specify a startHeight for the fork under which this bridge address config would be valid?

Reply:
good catch, this is redundant and should be inherited from the fork height. Fixed in commit 6de3a601136aa3eaa21164b48820659f4c6187e2